### PR TITLE
Bugfix/library fixes

### PR DESCRIPTION
**Pull request by crokita**
This PR is **ready** for review.

### Summary
The transport manager class has been split out so that implementation details for a transport are in a separate class. Also, the SdlSession and SdlProtocol related classes are fixed up so that a separate SdlSession extended class is no longer needed.

### CODE REVIEW

**Comment by crokita** [comment]
[(1.5) Example Project - Hello SDL Vanilla JS (register app, app icon, name, send show) [1.5]](https://trello.com/c/Tis8It0l/1322-15-example-project-hello-sdl-vanilla-js-register-app-app-icon-name-send-show-15)

File: lib/js/src/protocol/SdlProtocolBase.js [hunk 1]
```diff
@@ -70,10 +66,10 @@ class SdlProtocolBase {
         this._baseTransportConfig = baseTransportConfig;
         this._transportConfig = baseTransportConfig;
         this._sdlProtocolListener = sdlProtocolListener;
+        this._transportManager;
```
Comment (nickschwab) [inline lib/js/src/protocol/SdlProtocolBase.js]:
```suggestion
        this._transportManager = null;
```

File: lib/js/src/session/SdlSession.js [hunk 2]
```diff
@@ -77,7 +77,7 @@ class SdlSession {
 
         this._sdlProtocolListener = this._setupSdlProtocolListener();
 
-        this._sdlProtocol = this._setupSdlProtocol();
+        this._sdlProtocol = new SdlProtocol(baseTransportConfig, this._sdlProtocolListener);
     }
 
     _setupSdlProtocol() {
```
Comment (nickschwab) [inline lib/js/src/session/SdlSession.js]: Remove unused method

File: lib/js/src/transport/WebsocketClientTransport.js [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+import { TransportBase } from './TransportBase.js';
+import { SdlPsm } from './SdlPsm.js';
+
+class WebsocketClientTransport extends TransportBase {
+    constructor (config, transportListener) {
+        super(config, transportListener);
+        this._queue = [];
+        this._isRunning = false;
+        this._wsUrl = `${config.getHost()}:${config.getPort()}`;
+        this._ws;
```
Comment (nickschwab) [inline lib/js/src/transport/WebsocketClientTransport.js]:
```suggestion
        this._ws = null;
```

File: lib/js/src/transport/WebsocketClientTransport.js [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+        this._callback;
```
Comment (nickschwab) [inline lib/js/src/transport/WebsocketClientTransport.js]:
```suggestion
        this._callback = null;
```

**Pull request merged**